browser/resources/settings/brave_overrides/site_details.ts

```diff
@@ -3,52 +3,55 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this file,
 // you can obtain one at https://mozilla.org/MPL/2.0/.
 
-// @ts-nocheck TODO(petemill): Define types and remove ts-nocheck
+import type {SiteDetailsPermissionElement} from '../site_settings/site_details_permission.js'
 
 import {RegisterPolymerTemplateModifications} from 'chrome://resources/brave/polymer_overriding.js'
 import {html} from 'chrome://resources/polymer/v3_0/polymer/polymer_bundled.min.js'
-import {getTrustedHTML} from 'chrome://resources/js/static_types.js'
 import {loadTimeData} from '../i18n_setup.js'
 
 import 'chrome://resources/brave/leo.bundle.js'
 
 RegisterPolymerTemplateModifications({
   'site-details': (templateContent: HTMLTemplateElement) => {
     // Add top-padding to subpage
-    templateContent.prepend(html`<style>#usage { padding-top: var(--leo-spacing-l); }</style>`.content)
+    templateContent.prepend(
+      html`<style>#usage { padding-top: var(--leo-spacing-l); }</style>`.
+        content)
 
     if (!loadTimeData.getBoolean('isIdleDetectionFeatureEnabled')) {
-      const idleDetectionItem = templateContent.querySelector('[category="[[contentSettingsTypesEnum_.IDLE_DETECTION]]"]')
+      const idleDetectionItem =
+        templateContent.querySelector<SiteDetailsPermissionElement>(
+          '[category="[[contentSettingsTypesEnum_.IDLE_DETECTION]]"]')
       if (!idleDetectionItem) {
-        console.error(`[Brave Settings Overrides] Couldn't find idle detection item`)
+        console.error('[Settings] Couldn\'t find idle detection item')
       } else {
         idleDetectionItem.hidden = true
       }
     }
-    const adsItem = templateContent.querySelector('[category="[[contentSettingsTypesEnum_.ADS]]"]')
+    const adsItem =
+      templateContent.querySelector<SiteDetailsPermissionElement>(
+        '[category="[[contentSettingsTypesEnum_.ADS]]"]')
     if (!adsItem) {
-      console.error(`[Brave Settings Overrides] Couldn't find ads item`)
+      console.error('[Settings] Couldn\'t find ads item')
     } else {
       adsItem.hidden = true
     }
-    const firstPermissionItem = templateContent.querySelector('div.list-frame > site-details-permission:nth-child(1)')
+    const firstPermissionItem = templateContent.querySelector(
+      'div.list-frame > site-details-permission:nth-child(1)')
     if (!firstPermissionItem) {
-      console.error(`[Brave Settings Overrides] Couldn't find first permission item`)
+      console.error('[Settings] Couldn\'t find first permission item')
```

Comment:
```suggestion
      console.error("[Settings] Couldn't find first permission item")
```

Reply:
I think these are addressed in the "linter" commit.